test/general/key.js

```diff
@@ -2667,6 +2718,26 @@ describe('Key', function() {
     await expect(pubKey.getEncryptionKey()).to.be.rejectedWith('Could not find valid encryption key packet in key c076e634d32b498d');
   });
 
+  it('should not decrypt using a sign-only RSA key, unless explicitly configured', async function () {
+    const allowSigningKeyDecryption = openpgp.config.allow_insecure_decryption_with_signing_keys;
+    const { keys: [key] } = await openpgp.key.readArmored(rsaSignOnly);
+    console.log('off');
+    openpgp.config.allow_insecure_decryption_with_signing_keys = false;
+    await expect(openpgp.decrypt({
+      message: await openpgp.message.readArmored(encryptedRsaSignOnly),
+      privateKeys: key
+    })).to.be.rejectedWith(/Session key decryption failed/);
+    console.log('on');
+
+    openpgp.config.allow_insecure_decryption_with_signing_keys = true;
+    await expect(openpgp.decrypt({
+      message: await openpgp.message.readArmored(encryptedRsaSignOnly),
+      privateKeys: key
+    })).to.be.fulfilled;
+
+    openpgp.config.allow_insecure_decryption_with_signing_keys = allowSigningKeyDecryption;
```

Comment:
Please put the rest of this test in a `try` with this line in a `finally`, so that we don't mess up the config option if the test fails